dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/configuration/Configuration.kt

```diff
@@ -245,6 +247,19 @@ internal constructor(
             return this
         }
 
+        /**
+         * Sets the strategy to handle scalability issues.
+         * Many operations (data processing, event I/O, …) are queued in background threads.
+         * This configuration lets one decide how to handle the edge case when the queue starts growing, which can lead
+         * to a lot of memory usage, delayed processing, and possibly OOM or ANR.
+         * @param backpressureStrategy the backpressure strategy (default strategy ignores new tasks if a queue reaches
+         * 1024 items)
+         */
+        fun setBackpressureStrategy(backpressureStrategy: BackPressureStrategy): Builder {
```

Comment:
do we want to add this in the core or only for RUM for now? Except of RUM no other feature is using a dedicated executor service for the event processing (they all use I/O executor services for writing events, though), and only RUM has a single thread to pull work items from the queue, while other services have multiple threads (so, in theory, they should consume the queue faster).

Reply:
IMHO it makes more sense to make it a global configuration. A huge backlog in one executor can propagate to the persistence executor (which is what was experienced by our customer)